src/Azure.Deployments.Extensibility.Core/V2/Models/ModelSerializerContext.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
+
+namespace Azure.Deployments.Extensibility.Core.V2.Models
+{
+    [JsonSerializable(typeof(JsonObject))]
+    [JsonSerializable(typeof(ErrorData))]
+    [JsonSerializable(typeof(Resource))]
+    [JsonSerializable(typeof(ResourceSpecification))]
+    [JsonSerializable(typeof(ResourceReference))]
+    [JsonSerializable(typeof(ResourceOperation))]
```

Comment:
>JsonSerializable [](http://example.com/codeflow?start=5&length=16)

If we ever forget add a new class that also needs to marked here, are the errors pretty obvious?

Reply:
I think that depends on how the serializer context is used. If it is used directly then it is impossible to use it for a type that is not marked here. If it's used by ASP.NET core to serialize / deserialize requests and responses, I'm not sure if it will only fail if the app is configured to use AOT. I'd need to do some testing to find out. But if we ever forget to add a new class, the extension author can fix it by themselves, since they can decare a new serializer context and combine it with the one we provide.